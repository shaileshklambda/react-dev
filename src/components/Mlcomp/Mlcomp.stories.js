import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'
import Mlcomp from '.'

storiesOf('Mlcomp', module).add('default', () => <Mlcomp />)
