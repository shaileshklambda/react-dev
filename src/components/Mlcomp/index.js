// @flow
import React, { Component } from "react";
//import styles from "./index.module.css";
import HelpMenuComponent from "../HelpMenuComponent";
import AccesskeyPopover from "../AccessKeyPopover";
import EnvIcons from "../EnvIcons";
import Barchart from "../BarChart";
import VideoComponent from "../VideoComponent";
import DebuggerTab from "../DebuggerTab";
import TestDetails from "../TestDetails";
import Counter from "../Counter";
import { Row, Col } from "react-bootstrap";
import {
  chartPref,
  env,
  videoJsOptions,
  debuggerData,
  counterData
} from "../../common/default";
type Props = {
  sampleData: Object,
  envInfo: Object,
  chartPref: Object,
  debuggerData: Object,
  title: String,
  count: String
};
class Mlcomp extends Component<Props> {
  render() {
    return (
      <div>
        <Row>
          <Col md={4}>
            <HelpMenuComponent />
          </Col>
          <Col md={4}>
            <AccesskeyPopover />
          </Col>
          <Col md={4}>
            <EnvIcons
              envInfo={this.props.envInfo ? this.props.envInfo : null}
            />
          </Col>
          <Col md={8}>
            {this.props.chartPref &&
            Object.keys(this.props.chartPref).length > 0 ? (
              <Barchart
                pref={this.props.chartPref ? this.props.chartPref : {}}
              />
            ) : null}
          </Col>
        </Row>
        <Row>
          <Col md={3}>
            <VideoComponent {...this.props.videoOptions} />
          </Col>
          <Col md={6}>
            <DebuggerTab {...this.props.debuggerData} />
          </Col>
        </Row>
        <Row>
          <Col md={6}>
            <TestDetails />
          </Col>
          <Col md={6}>
            <Counter counter={this.props.counterData} />
          </Col>
        </Row>
      </div>
    );
  }
}
Mlcomp.defaultProps = {
  envInfo: env,
  chartPref: chartPref,
  videoOptions: videoJsOptions,
  debuggerData: debuggerData,

  counterData: counterData
};
export default Mlcomp;
