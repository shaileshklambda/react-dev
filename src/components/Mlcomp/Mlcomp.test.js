import React from "react";
import renderer from "react-test-renderer";
import Mlcomp from ".";

it("Mlcomp: default", () => {
  let sampleData = {};
  let envInfo = {};
  let chartPref = {};
  const component = renderer.create(
    <Mlcomp sampleData={sampleData} envInfo={envInfo} chartPref={chartPref} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
