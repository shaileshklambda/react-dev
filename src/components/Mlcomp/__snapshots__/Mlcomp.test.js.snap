// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mlcomp: default 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    >
      <div>
        <img
          className=""
          height={20}
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "help-center.svg",
            }
          }
          width={25}
        />
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div>
        <img
          className=""
          height={20}
          onClick={[Function]}
          src={
            Object {
              "ReactComponent": [Function],
              "__esModule": true,
              "default": "access-key.svg",
            }
          }
          width={25}
        />
      </div>
    </div>
    <div
      className="col-md-4"
    >
      <div
        className="default"
      />
    </div>
    <div
      className="col-md-8"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-3"
    >
      <div>
        <div
          data-vjs-player={true}
        >
          <video
            className="video-js vjs-big-play-centered"
            poster=""
          />
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div
        className="colAdjust col-md-12 col-xs-12"
      >
        <div
          className="Debuggertab"
        >
          <div
            className="bug_list_tab activeParent"
            id="uncontrolled-tab-example"
          >
            <ul
              className="nav nav-tabs"
              role="tablist"
            />
            <div
              className="tab-content"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-6"
    >
      <div
        className="col-md-12 col-xs-12"
      >
        <div>
          <div
            className="row"
          >
            <div
              className="col-md-7 col-xs-7"
            >
              <div
                className="row"
              >
                <div
                  className="headingParent col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-12 col-xs-12"
                  >
                    <span
                      className="testHeading"
                    >
                      Falcon/Realtime Browser
                    </span>
                    <span
                      className="highlightedText"
                    >
                       
                       FS-006S
                    </span>
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          Tester: 
                        </strong>
                        Gragnama
                      </p>
                    </span>
                  </div>
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          CI Tool: 
                        </strong>
                         
                        Jenkins
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  className="col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          Local Tunnel: 
                        </strong>
                         
                        Null
                      </p>
                    </span>
                  </div>
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          No. of Issues: 
                        </strong>
                         
                        20
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  className="col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          Start Time: 
                        </strong>
                         
                        16:00
                      </p>
                    </span>
                  </div>
                  <div
                    className="col-md-6 col-xs-6"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          End Time: 
                        </strong>
                         
                        20:00
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  className="col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-12 col-xs-12"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          Base Url: 
                        </strong>
                         
                        https://www.lambdatest.com
                      </p>
                    </span>
                  </div>
                </div>
                <div
                  className="col-md-12 col-xs-12"
                >
                  <div
                    className="col-md-12 col-xs-12"
                  >
                    <span
                      className="innerContent"
                    >
                      <p>
                        <strong>
                          Environment: 
                        </strong>
                         
                        <img
                          className=""
                          src="icons/windows.png"
                        />
                         
                        10
                         
                        |
                         
                        <img
                          className=""
                          src="icons/google_chrome.png"
                        />
                         
                        69.0
                         
                        |
                         
                        <img
                          className=""
                          src="icons/computer.png"
                        />
                         
                        1900 x 1080
                      </p>
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-md-4 col-xs-4"
            >
              <div>
                <div
                  data-vjs-player={true}
                >
                  <video
                    className="video-js vjs-big-play-centered"
                    poster=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-md-6"
    >
      <div>
        <div
          className="row"
        >
          <div
            className="col-md-4"
          >
            <div
              className="rectangle"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-12"
                >
                  <span
                    className="title"
                  >
                    Total Bugs Logged
                  </span>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="col-md-2"
                >
                  <span
                    className="count"
                  >
                    29
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
